Feature new user form page

ArturasSosnovskis opened the pull request:
new user registration form, awaiting comments about the code

kazenaite7 committed:
Merge pull request #3 from NeHerkus/stable

Fixed

Commit:
tidied up new idea formpage .css file and some vars in .html

Created new user registration form page with basic field validations
(more validation comming soon)

Commit:
fixed new-user-formpage test

gpiliponyte commented on src/app/newuserformpage/new-user-formpage.component.ts:
```diff
@@ -0,0 +1,37 @@
+  initForm() {
+    this.userForm = this.formBuilder.group({
+      fullName: ['', [Validators.required, Validators.maxLength(this.maxInputLength77)]],
+      email: [this.formControl, Validators.maxLength(this.maxInputLength77)],
```
why did you separate the email validators? :D

gpiliponyte replied:
email: ['', [
    Validators.required,
    Validators.email, Validators.maxLenth(...)]] should work fine

gpiliponyte commented on src/app/app-routing.module.ts:
```diff
@@ -3,12 +3,14 @@ import {Routes, RouterModule} from '@angular/router';
   {path: 'home', component: HomePageComponent},
   {path: 'newidea', component: NewIdeaFormpageComponent},
+  {path: 'newuser', component: NewUserFormpageComponent},
   {path: '', redirectTo: '/home', pathMatch: 'full'},
```
this is not really needed, since the wildcard already redirects to home

gpiliponyte commented on src/app/app-routing.module.ts:
```diff
@@ -3,12 +3,14 @@ import {Routes, RouterModule} from '@angular/router';
 
 const routes: Routes = [
   {path: 'home', component: HomePageComponent},
   {path: 'newidea', component: NewIdeaFormpageComponent},
```
NewIdeaFormPageComponent

gpiliponyte commented on src/app/newuserformpage/new-user-formpage.component.html:
```diff
@@ -0,0 +1,73 @@
+
+<form class="form" [formGroup]="userForm">
+  <mat-card class="card">
+    <h2 style="text-align: center">Please fill the user registration form</h2>
```
still some inline styling left :)

gpiliponyte commented on src/app/newuserformpage/new-user-formpage.component.ts:
```diff
@@ -0,0 +1,37 @@
+  styleUrls: ['./new-user-formpage.component.css']
+})
+export class NewUserFormpageComponent implements OnInit {
+  maxInputLength77 = 77;
```
not what I mean when I said you should move things like that to constants. 
It would be good to create a separate file for constants, then store the constants like this 
export const userFormOptions = {
fullNameMaxLength: 77,
emailMaxLength: 77
}

gpiliponyte replied:
The same should be done with the constants in the idea form

gpiliponyte commented on src/app/newuserformpage/new-user-formpage.component.html:
```diff
@@ -0,0 +1,73 @@
+<form class="form" [formGroup]="userForm">
+  <mat-card class="card">
+    <h2 style="text-align: center">Please fill the user registration form</h2>
+    <mat-card-content style="height: auto">
```
inline styling

gpiliponyte commented on src/app/newuserformpage/new-user-formpage.component.html:
```diff
@@ -0,0 +1,73 @@
+        <input matInput
+               type="text"
+               formControlName="fullName"
+               maxlength="77"
```
use constants

gpiliponyte commented on src/app/newideaformpage/new-idea-formpage.component.ts:
```diff
@@ -69,7 +69,6 @@ export class NewIdeaFormpageComponent implements OnInit {
       this.ideaService.createIdea(this.ideaForm.value).subscribe(
         res => {
           console.log('Request succesfully sent');
-          // location.reload();
```
no comments guys :D

gpiliponyte commented on src/app/newuserformpage/user-form-error-state-matcher.ts:
```diff
@@ -0,0 +1,9 @@
+import {ErrorStateMatcher} from '@angular/material';
+import {FormControl, FormGroupDirective, NgForm} from '@angular/forms';
+
+export class UserFormErrorStateMatcher implements ErrorStateMatcher {
```
doesn't seem like this is necessary?

gpiliponyte replied:
the email field should work fine without it

NeHerkus committed:
Add "Page not found" handle

NeHerkus reviewed (approved):
Minor fixes needed, but nothing major. Fix constants 🐞

NeHerkus committed:
Merge pull request #7 from NeHerkus/feature-page-not-found

Add "Page not found" handle

Commit:
Changed input fields validation logic
added further custom validation (email,password fields)
tidied up userForm formGroup

Commit:
Merge branch 'stable' of github.com:NeHerkus/good-deeds-front into feature-new-user-form-page

# Conflicts:
#	src/app/app.module.ts

Commit:
Merge branch 'stable' of github.com:NeHerkus/good-deeds-front into feature-new-user-form-page

Commit:
travis test fix

Commit:
routing added form newuser

Commit:
routing added form newuser

gpiliponyte commented on src/app/newideaformpage/new-idea-formpage.component.ts:
```diff
@@ -10,6 +10,13 @@ import {MatSnackBar} from '@angular/material';
+  private descFieldMaxSymbols = 500;
+ private contactPersFieldMaxSymbols = 150;
   ideaForm: FormGroup;
   locations: string[] = [
```
I think this should be an enum

gpiliponyte commented on src/app/newideaformpage/new-idea-formpage.component.ts:
```diff
@@ -10,6 +10,13 @@ import {MatSnackBar} from '@angular/material';
+  private websiteFieldMaxSymbols = 77;
+  private optPartFieldMaxSymbols = 20;
+  private descFieldMaxSymbols = 500;
+ private contactPersFieldMaxSymbols = 150;
```
export const newIdeaFormOptions = {
 nameFieldMaxSymbols = 77,
 locationFieldMaxSymbols = 77,
 ....ect
}

Commit:
fixing travis errors
